From 2819311d6e2151fd25a2d66fb3513341d0f7da05 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lubom=C3=ADr=20Sedl=C3=A1=C5=99?= <lsedlar@redhat.com>
Date: Wed, 18 Oct 2017 15:22:41 +0200
Subject: [PATCH] config: Allow comps_file for any gather_source
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

It only affects gathering packages when gather_source is set to comps,
but it could still make sense to include the file in the repository even
if the not used for gathering.

Signed-off-by: Lubomír Sedlář <lsedlar@redhat.com>
---
 pungi/checks.py      | 1 -
 tests/test_config.py | 3 +--
 2 files changed, 1 insertion(+), 3 deletions(-)

diff --git a/pungi/checks.py b/pungi/checks.py
index d9f82c1..acb3f4b 100644
--- a/pungi/checks.py
+++ b/pungi/checks.py
@@ -1138,7 +1138,6 @@ CONFIG_DEPS = {
     "gather_source": {
         "conflicts": [
             (lambda val: val != 'json', ['gather_source_mapping']),
-            (lambda val: val != 'comps', ['comps_file']),
         ],
         "requires": [
             (lambda val: val == 'json', ['gather_source_mapping']),
diff --git a/tests/test_config.py b/tests/test_config.py
index 24bc72d..5f7d04b 100644
--- a/tests/test_config.py
+++ b/tests/test_config.py
@@ -247,8 +247,7 @@ class GatherConfigTestCase(ConfigTestCase):
 
         self.assertValidation(
             cfg,
-            [checks.REQUIRES.format('gather_source', 'json', 'gather_source_mapping'),
-             checks.CONFLICTS.format('gather_source', 'json', 'comps_file')])
+            [checks.REQUIRES.format('gather_source', 'json', 'gather_source_mapping')])
 
 
 class OSBSConfigTestCase(ConfigTestCase):
-- 
2.13.6

